model_analyzer/config/generate/quick_plus_concurrency_sweep_run_config_generator.py

```diff
@@ -0,0 +1,188 @@
+                 config: ConfigCommandProfile, gpus: List[GPUDevice],
+                 models: List[ConfigModelProfileSpec], client: TritonClient,
+                 result_manager: ResultManager,
+                 model_variant_name_manager: ModelVariantNameManager):
+        """
+        Parameters
+        ----------
+        search_config: SearchConfig
+            Defines parameters and dimensions for the search
+        config: ConfigCommandProfile
+            Profile configuration information
+        gpus: List of GPUDevices
+        models: List of ConfigModelProfileSpec
+            List of models to profile
+        client: TritonClient
+        result_manager: ResultManager
+            The object that handles storing and sorting the results from the perf analyzer
+        model_variant_name_manager: ModelVariantNameManager
+            Maps model variants to config names
+        
+        model_variant_name_manager: ModelVariantNameManager
+        """
+        self._search_config = search_config
+        self._config = config
+        self._gpus = gpus
+        self._models = models
+        self._client = client
+        self._result_manager = result_manager
+        self._model_variant_name_manager = model_variant_name_manager
+        self._rcg = None
+
+    def set_last_results(self, measurements):
+        self._rcg.set_last_results(measurements)
+
+    def get_configs(self) -> Generator[RunConfig, None, None]:
+        """
+        Returns
+        -------
+        RunConfig
+            The next RunConfig generated by this class
+        """
+        yield from self._execute_quick_search()
+        yield from self._sweep_concurrency_over_top_results()
+
+    def _execute_quick_search(self):
+        self._rcg = self._create_quick_run_config_generator()
+
+        yield from self._rcg.get_configs()
+
+    def _create_quick_run_config_generator(self) -> QuickRunConfigGenerator:
+        return QuickRunConfigGenerator(
+            search_config=self._search_config,
+            config=self._config,
+            gpus=self._gpus,
+            models=self._models,
+            client=self._client,
+            model_variant_name_manager=self._model_variant_name_manager)
+
+    def _sweep_concurrency_over_top_results(self):
+        top_results = self._result_manager.top_n_results(
+            n=DEFAULT_NUM_CONFIGS_PER_MODEL)
+
+        for count, result in enumerate(top_results):
+            new_config = self._create_new_config_command_profile(result)
+            self._rcg = self._create_brute_run_config_generator(
+                new_config, skip_default_config=(count != 0))
+
+            yield from self._rcg.get_configs()
+
+    def _create_new_config_command_profile(
+            self, result: RunConfigResult) -> ConfigCommandProfile:
+        new_config = deepcopy(self._config)
+
+        self._set_search_mode(new_config)
+        self._set_parameters(result, new_config)
```

Comment:
I don't love that result is an input, and new_config is an input and output. Can we make them all return config so the code is more obvious what is being modified?

Reply:
Done